docs/dev-client-gameplay.md

```diff
@@ -1,161 +1,333 @@
+eg: `(world: HubsWorld) => void`.
+
+
+### The game loop
+
+We provide the browser&rsquo;s
+[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)
+(or [`XRSession.requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/XRSession/requestAnimationFrame)
+in [`immersive mode`](https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API))
+with our game loop function (`mainTick`), to be invoked each frame. All the
+systems are invoked from `mainTick` one by one.
 
-`bitECS` queries allow us to find entities based on the components that are attached to them. `enterQuery` and `exitQuery` wrap regular queries so that we handle when an entity first matches or stops matching a given query. The `bitECS` documentation should be consulted for more details.
+If you want to your addon you need to edit `mainTick` to insert your systems.
 
+TODO: Write about system order
 
-<a id="org2233e62"></a>
 
-## Replacing `async` functions with `coroutines`
+### Queries are useful
 
-TODO: Describe `JobRunner` and `coroutine` s.
+[`bitECS` queries](https://github.com/NateTheGreatt/bitECS/blob/master/docs/INTRO.md#-query)
+allow us to find entities based on the [components](https://github.com/NateTheGreatt/bitECS/blob/master/docs/INTRO.md#-component)
+that are attached to them. `enterQuery` and `exitQuery` wrap regular queries so
+that we handle when an entity first matches or stops matching a given query.
+[The `bitECS` documentation](https://github.com/NateTheGreatt/bitECS#--documentation)
+should be consulted for more details.
 
+Holding the references to entities may be danger because entities can be
+deleted or even recycled [as explained later](#creating-entities). It is safer
+to use queries as much as possible to fetch entities when needed.
 
-<a id="org7331485"></a>
 
-# Writing components
+### Coroutines
 
+`async` functions should not be used. Instead, `coroutines` should be used to
+ensure that systems run in `mainTick`.
 
-<a id="org18a6d0b"></a>
+If you use `async` in a system, the system can restart outside of `mainTick`.
+Using `coroutines` enforces systems to run in `mainTick`. We want all the
+systems to run in `mainTick` for good lifecycle control.
 
-## Defining components
+We provide `coroutine` helper functions. Refer to [the "JobRunner" section](#jobrunner)
+below for details.
 
-`bitECS` components are defined with `defineComponent`.
 
+## Writing components
 
-<a id="org3db6503"></a>
 
-## Data types
+### Defining components
 
-`bitECS` components only store numeric types: `i8`, `ui8`, `ui8c`, `i16`, `ui16`, `i32`, `ui32`, `f32`, `f64`, and `eid`.
+`bitECS` [components](https://github.com/NateTheGreatt/bitECS/blob/master/docs/INTRO.md#-component)
+are defined with [`defineComponent`](https://github.com/NateTheGreatt/bitECS/blob/master/docs/API.md#definecomponent--object).
 
-The sections below describe what we do when we need to store non-numeric data.
 
+### Data types

```

Comment:
This has the same hierarchical level as the sections that it presumably contains (References, Strings, Flags, The escape hatch). Perhaps those lower level sections could be h4’s. I’m unsure of how this will look in Github md renderer.